src/core/mac/sub_mac.cpp

```diff
@@ -227,6 +240,32 @@ void SubMac::StartCsmaBackoff(void)
     uint32_t backoff;
     uint32_t backoffExponent = kMinBE + mTransmitRetries + mCsmaBackoffs;
 
+#if OPENTHREAD_CONFIG_CSL_TRANSMITTER_ENABLE
+    if (mTransmitFrame.mInfo.mTxInfo.mPeriod != 0)
+    {
+        uint32_t phaseNow     = (otPlatTimeGet() / OT_US_PER_TEN_SYMBOLS) % mTransmitFrame.mInfo.mTxInfo.mPeriod;
+        uint32_t phaseDesired = mTransmitFrame.mInfo.mTxInfo.mPhase;
```

Comment:
Rather than storing `mPhase` and `mPeriod` in `mTxInfo`, how about just store the desired transmit time in `mTxInfo` to avoid recalculate again here?

Reply:
But it needs to handle the case where the desired transmit time is earlier than current time. So even if we calculate the desired time in advance, we still need to compare it with current time and do some handling here.